docs/source/tutorial/checkpoints.rst

```diff
@@ -0,0 +1,147 @@
+setting the argument ``checkpoint_frequency`` with an integer. The default frequency is 30 minutes and setting it to
+``0`` will cause the training loop to save a checkpoint after each epoch.
+
+Here is an example:
+
+.. code-block:: python
+
+    losses = training_loop.train(
+        num_epochs=1000,
+        checkpoint_name='my_checkpoint.pt',
+        checkpoint_frequency=5,
+    )
+
+With this code we have started the training loop with the above defined KGEM. The training loop will save a checkpoint
+in the ``my_checkpoint.pt`` file, which will be saved in the ``~/.pykeen/checkpoints/`` directory, since we haven't
+set the argument ``checkpoint_directory``.
+The checkpoint file will be saved after 5 minutes since starting the training loop or the last time a checkpoint was
+saved and the epoch finishes, i.e. when one epoch takes 10 minutes the checkpoint will be saved after 10 minutes.
+In addition, checkpoints are always saved when the early stopper stops the training loop or the last epoch was finished.
+
+Let's assume you were anticipative, saved checkpoints and your training loop crashed after 200 epochs.
+Now you would like to resume from the last checkpoint. All you have to do is to rerun the **exact same code** as above
+and PyKEEN will smoothly start from the given checkpoint. Since PyKEEN stores all random states as well as the
+states of the model, optimizer and early stopper, the results will be exactly the same compared to running the
+training loop uninterruptedly. Of course, PyKEEN will also continue saving new checkpoints even when
+resuming from a previous checkpoint.
+
+On top of resuming interrupted training loops you can also resume training loops that finished successfully.
+E.g. the above training loop finished successfully after 1000 epochs, but you would like to
+train the same model from that state for 2000 epochs. All you have have to do is to change the argument
+``num_epochs`` in the above code to:
+
+.. code-block:: python
+
+    losses = training_loop.train(
+        num_epochs=2000,
+        checkpoint_name='my_checkpoint.pt',
+        checkpoint_frequency=5,
+    )
+
+and now the training loop will resume from the state at 1000 epochs and continue to train until 2000 epochs.
+
+Another nice feature is that the checkpoints functionality integrates with the pipeline. This means that you can simply
+define a pipeline like this:
+
+.. code-block:: python
+
+    from pykeen.pipeline import pipeline
+    pipeline_result = pipeline(
+        dataset='Nations',
+        model='TransE',
+        optimizer='Adam',
+        training_kwargs=dict(num_epochs=1000, checkpoint_name='my_checkpoint.pt', checkpoint_frequency=5),
+    )
+
+Again, assuming that e.g. this pipeline crashes after 200 epochs, you can simply execute **the same code** and the
+pipeline will load the last state from the checkpoint file and continue training as if nothing happened.
+
+.. todo:: Tutorial on recovery from hpo_pipeline.
```

Comment:
Maybe for a later PR?

Reply:
Yes. Basically the hop_pipeline supports saving checkpoints through the pipeline, but that is specific to the training loop itself. Supporting to resume a cancelled hpo_pipeline would be an entirely different story.

Reply:
I wonder if this `.. todo` directive breaks the build